### ISSUE TITLE
Unable to access mounted drives (NAS) in Windows

### ISSUE THREAD

**Issue by defeatedbythecat** [jellyfin/jellyfin#1836]
**Describe the bug**
During setup, the network box returns no entries and jellyfin server see's nothing but the C: drive. Despite there being other mapped drives. 
**System**
 - OS:  Windows
 - Browser:  Chrome
 - Jellyfin Version: Latest nightly since Sept 20

**Additional context**
Using a Synology NAS containing the media.

**Comment by anthonylavado** [comment]
@defeatedbythecat Are you using the Installer version of Jellyfin?

**Comment by defeatedbythecat** [comment]
Yeah using the new installer

**Comment by anthonylavado** [comment]
Okay. So this is a side effect of the way the installer sets up Jellyfin with the “Network Service” account. It likely doesn’t have permissions to access your shared drives. @EraYaN had a better explanation, but I forget what it is.

**Comment by HansGruberXXV** [comment]
I'm having the same issue. I've set up permissions, i think. Added Network Service to Users  and I get an message box saying that it couldn't start the service but that it didn't throw an error. This is using the most recent download of 10.4 x64

**Comment by EraYaN** [comment]
Network Service (and Local System) for that matter can only access network shares or mounted drives in a Windows Domain environment. Since they identify themselves as the computer account on the network. So outside a domain the target host has no idea about the client host and can't give it access.

Your best bet is to create a new user that is named (for example `Jellyfin`) and use the `net use` and `runas` commands to give that user the credentials to the share. Then Jellyfin should be able to enumerate the shares.

I suppose the other option is to do full public samba shares, but that is very insecure. (It's guest access in the samba config) That should work, although I'm not 100% on that.

**Comment by HansGruberXXV** [comment]
All right, I think I'm in a little over my head here. I have the drives mapped as Z:\ for movies and Y:\ for TV. I'd prefer to leave them in their current state as multiple programs use them for downloading, sorting, cataloguing etc. I've created the user jellyfin and added it to the users group and then made sure it has access permission to the folder that jellyfin.exe is located so that I can use that account to run as a service. 

Where I would appreciate some guidance is how to add the network share access to the jellyfin user account. I've looked at net use but as the drives are already mapped it tells me it can't map them for another user. Runas seems to be looking for an exe to run, unless I'm mistaken about it's usage. I'm sorry to be a pain, I feel super close to having the damn thing work, haha.

**Comment by EraYaN** [comment]
Well yes those drives are user based, and mapping to multiple users is impossible. It might be the easiest to just use netuse to provide the credentials for the sharename directly. And use that, I believe jellyfin should be able to work with UNC paths. runas can be used to run net use as the jellyfin user without actually login in to that user, although using it can be a bit finicky, but think of it as `sudo -Hu`

**Comment by defeatedbythecat** [comment]
Is this something that might be worked out in future releases, or is it an inherent issue with the way windows works?

I know it's a completely different piece of software, but shouldn't Plex suffer from a similar issue if it's a windows problem?

**Comment by anthonylavado** [comment]
>Is this something that might be worked out in future releases, or is it an inherent issue with the way windows works? ...

@defeatedbythecat It’s an inherent issue with the way Windows works - when you’re running as a service. Plex and Emby do not run as a service, they require you to login and start/auto-start the app. This feature is in the works.

**Comment by krizly** [comment]
Possibly related to this issue but not sure if maybe there's something else going on -
I have just setup a clean install of Jellyfin 10.4.0 on a new install of Windows 10 which is connected to a domain and installed using a network service account. Signed in to Windows10 with UserX, changed the Jellyfin service to run as UserX. Have mapped drives to D and Z under UserX, but neither show up when trying to add folders to a library, and trying to manually add the server path in the 'Shared Network Folder' field fails. 

Also tried it just as the Network Service account and added the computer AD account to the network share group so it had access, but still couldn't access the network shares from Jellyfin.

**Comment by EraYaN** [comment]
Mmm if you enter the full UNC path in the main path operation and all computers are on the same AD domain AND the computer account has access to both the share AND the NTFS files, theoretically this should mean jellyfin should be able to read those files.

Also I think mapped drives might not get mounted when using services (although it marked to be remembered they should)
Also the "Shared Network Folder" option is the path to be passed to certain clients to access the same files (Kodi mostly) but you also need to enter that same network path in the main path selector.

**Comment by magicalbanana** [comment]
> Network Service (and Local System) for that matter can only access network shares or mounted drives in a Windows Domain environment. Since they identify themselves as the computer account on the network. So outside a domain the target host has no idea about the client host and can't give it access.
> 
> Your best bet is to create a new user that is named (for example `Jellyfin`) and use the `net use` and `runas` commands to give that user the credentials to the share. Then Jellyfin should be able to enumerate the shares.
> 
> I suppose the other option is to do full public samba shares, but that is very insecure. (It's guest access in the samba config) That should work, although I'm not 100% on that.

Can you perhaps give a tutorial on how to achieve this? I tried `net use` but I don't think I'm doing it right.

**Comment by fthorsen76** [comment]
> > Network Service (and Local System) for that matter can only access network shares or mounted drives in a Windows Domain environment. Since they identify themselves as the computer account on the network. So outside a domain the target host has no idea about the client host and can't give it access.
> > Your best bet is to create a new user that is named (for example `Jellyfin`) and use the `net use` and `runas` commands to give that user the credentials to the share. Then Jellyfin should be able to enumerate the shares.
> > I suppose the other option is to do full public samba shares, but that is very insecure. (It's guest access in the samba config) That should work, although I'm not 100% on that.
> 
> @EraYaN, can you perhaps give a tutorial on how to achieve this? I tried `net use` but I don't think I'm doing it right.

I just installed a new Windows Server 2022 VM, and was struggling with this same issue. This is how I solved it;
1. If you don't have a dedicated (non-admin) user to access your media shares, then create one on your NAS and give it the correct permissions. Lets assume the username is JellyMedia
2. Regardless of options during the installation of Jellyfin, in Windows create a user account with the same account name and password as you did on the NAS (so JellyMedia). For now this user must also be a member of the local Adminsitrators group for this configuration to work! :(
3. In Windows, open Services, and locate the **Jellyfin Server** service. Change the **log on as** to the account you created in step 2. Then restart the service.
4. When adding a library in Jellyfin, only use the **Folder** option, NOT the shared network folder. And you must use UNC path, not mapped drives. So for instance use \\nas01\media\movies

That should do it. Now I only have to figure out what rights the service accounts needs so that it doesn't need to run as a local admin user. Right now the service throws an error 143 in the event log. The account automatically is granted the log on as a service right when adding it in step 3, so it must be something else.
Update; just as I wrote that, an idea came to mind, and it worked! It's the NTFS rights! So, do the following;
Navigate so you can right click the installation folder for Jellyfin (C:\Program Files\Jellyfin) and select Properties. Go to Security and either **Edit** the rights for the Jellyfin user in step 2, or for the _Users_ group and grant the **Modify** right.

**Comment by anthonylavado** [comment]
@fthorsen76 Thanks for your comments on this. I don't run it extensively on Windows, but as the maintainer of the tray app/installer, this is useful for me to add to the documentation. It'll help add to the installer to make sure things are as smooth as possible for new users.

**Comment by tenntenntenn** [comment]
Hello - just wanted to add to this (apologies if its not the correct place, new to jellyfin, github, etc):

I have a storage server with all my content on Windows Server 2019 Core ("STORAGE"), and a media server on Windows Server 2019 (non-core) ("MEDIA"), non-domain. I created a service account on both with the same username/password. I made sure that MEDIA can reach the files on STORAGE. I installed Jellyfin on MEDIA, set it's service to run as the service account, started it up, went through the wizard, got stuck at trying to add the path to my content on STORAGE. Was initially trying:

> \\STORAGE\media\movies

...didn't work, looked it up, found that Jellyfin can't use that, and the general guidance I found was to map a network drive on the Jellyfin server. So I did so, mapping the location \\STORAGE\media to M:\, and making sure that the drive was under the service account and something else (like the local administrator). I confirmed I could see and access the drive in Windows in Network Locations. So I then tried:

> M:\movies

Did not work. So I dug around a bunch, tried a few different things, tried reinstalling Jellyfin (note on a separate issue I found on that below), until I found this thread and the above comment by @fthorsen76 . So I tried:

> /STORAGE/media/movies

...didn't work. So then tried:

> //STORAGE/media/movies

...and that worked.

To note (sorry for being off topic): on reinstall, Jellyfin wouldn't run through the first time install process of creating a new user. It asks me to select my server (which I was already navigated to in my browser), then asks for a username/password. The account I created before the reinstall did not work, so I did not have a way to get in. I chose the option during the uninstall to remove local data, and confirmed it was gone in C:\programdata\jellyfin (and also poked around a few other common locations in Windows). Uninstalled, restarted the server, confirmed old data was gone, reinstalled, same issue. Found a guide on resetting this, by going to C:\ProgramData\Jellyfin\Server\config\system, and setting `<IsStartupWizardCompleted>true</IsStartupWizardCompleted>` to false.... but it was already false. So I stopped the service, set it to true, started the service, then stopped the service, set it to false, and started the service, and it then brought me through the first time startup again.

**Comment by ArtrixTech** [comment]
> > > Network Service (and Local System) for that matter can only access network shares or mounted drives in a Windows Domain environment. Since they identify themselves as the computer account on the network. So outside a domain the target host has no idea about the client host and can't give it access.
> > > Your best bet is to create a new user that is named (for example `Jellyfin`) and use the `net use` and `runas` commands to give that user the credentials to the share. Then Jellyfin should be able to enumerate the shares.
> > > I suppose the other option is to do full public samba shares, but that is very insecure. (It's guest access in the samba config) That should work, although I'm not 100% on that.
> > 
> > 
> > @EraYaN, can you perhaps give a tutorial on how to achieve this? I tried `net use` but I don't think I'm doing it right.
> 
> I just installed a new Windows Server 2022 VM, and was struggling with this same issue. This is how I solved it;
> 
> 1. If you don't have a dedicated (non-admin) user to access your media shares, then create one on your NAS and give it the correct permissions. Lets assume the username is JellyMedia
> 2. Regardless of options during the installation of Jellyfin, in Windows create a user account with the same account name and password as you did on the NAS (so JellyMedia). For now this user must also be a member of the local Adminsitrators group for this configuration to work! :(
> 3. In Windows, open Services, and locate the **Jellyfin Server** service. Change the **log on as** to the account you created in step 2. Then restart the service.
> 4. When adding a library in Jellyfin, only use the **Folder** option, NOT the shared network folder. And you must use UNC path, not mapped drives. So for instance use \nas01\media\movies
> 
> That should do it. Now I only have to figure out what rights the service accounts needs so that it doesn't need to run as a local admin user. Right now the service throws an error 143 in the event log. The account automatically is granted the log on as a service right when adding it in step 3, so it must be something else.

That worked for me. Thanks!

**Comment by ormissia** [comment]
> 

Very useful

**Comment by fernvenue** [comment]
Same here, and I just fixed and tested it on version 10.8.10, that's the latest version when I write this, let me explain how I did it, hope this can help others.

As @fthorsen76 mentioned, same username and password is required, if not, change it first. And if you run Jellyfin as system service, you may also need to change Jellyfin service's `Log On As` account to your specific user instead of `NT Authority\NetworkService`. To do that, you can run CMD (may or may not as Administrator) and run this:

```
cd "C:\Program Files\Jellyfin\Server"
nssm edit "Jellyfin Server"
```

Now you can see NSSM service editor user interface, go to `Log On` and change the user to your specific user. By the way, here I used default path, if you got any error you can find your nssm location by yourself.

Then, press `WIN` and type `services` (or any other way) to go Windows Services pannel, find `Jellyfin Server` and restart it, after that, your Jellyfin should run as your specific user.

And now, create a symbol link to local folder, here's an example:

```
mklink /d "X:\FOLDER_NAME" "\\SMB_SERVER_ADDR\YOUR_USERNAME_HERE\REMOTE_TARGET_PATH"
```

Let's say if our username is `bob`, and our SMB server is working on `192.0.2.1`, and we have some media files on it like `\\192.0.2.1\bob\media`, now if we wanna access it directly from `D:\Media`, we can do this:

```
mklink /d "D:\Media" "\\192.0.2.1\bob\media"
```

Or, we can directly access our whole SMB server like:

```
mklink /d "D\NAS" "\\192.0.2.1\bob"
```

Choose one way you like it :)

Finally, now we can go Jellyfin dashboard to add these local symbol folder, that's exactly the same as adding a real local folder.

Just my personal way to fix this, hope this can help others.
